src/rred_reports/reports/schools.py

```diff
@@ -0,0 +1,222 @@
+        """
+        return school_df
+
+    def get_outcome_from_summary(school_df, outcome_type):
+        try:
+            return school_df["exit_outcome"].value_counts()[outcome_type]
+        except KeyError:
+            return 0
+
+    filtered = filter_summary_table(school_df)
+    filtered_summary_table = filtered[columns_used]
+
+    return pd.DataFrame(
+        {
+            "number_of_rr_teachers": [filtered_summary_table["rred_user_id"].nunique()],
+            "number_of_pupils_served": [filtered_summary_table["pupil_no"].nunique()],
+            "po_discontinud": get_outcome_from_summary(filtered_summary_table, "Discontinued"),
+            "po_referred_to_school": get_outcome_from_summary(filtered_summary_table, "Referred to school"),
+            "po_incomplete": get_outcome_from_summary(filtered_summary_table, "Incomplete"),
+            "po_left_school": get_outcome_from_summary(filtered_summary_table, "Left school"),
+            "po_ongoing": get_outcome_from_summary(filtered_summary_table, "Ongoing"),
+        }
+    )
+
+
+def populate_school_tables(school_df: pd.DataFrame, template_path: Path) -> TemplateFiller:
+    """Function to fill the school template tables, saving them the file
+
+    Args: school_filter(pd.DataFrame), Path, school_id
+
+    Returns: The template filler with populated data
+    """
+
+    # adding a column for table four
+    lost_lesson_cols = [col for col in school_df if col.startswith("exit_lessons_missed")]
+    school_df["total_lost_lessons"] = school_df[lost_lesson_cols].sum(axis=1)
+
+    columns_and_filters = (
+        (table_one_columns, filter_one),
+        (table_two_columns, filter_two),
+        (table_three_columns, filter_three),
+        (table_four_columns, filter_four),
+        (table_five_columns, filter_five),
+        (table_six_columns, filter_six),
+    )
+
+    header_rows = [2, 1, 1, 1, 1, 2, 2]
+
+    # adding in summary table first
+    template_filler = TemplateFiller(template_path, header_rows)
+    add_in_summary_table = summary_table(school_df)
+    template_filler.populate_table(0, add_in_summary_table)
+
+    # now adding other tables
+    for index, column_and_filter in enumerate(columns_and_filters):
+        columns, filter_function = column_and_filter
+        filtered = filter_function(school_df)
+        table_to_write = filtered[columns]
+        template_filler.populate_table(index + 1, table_to_write)
+
+    return template_filler
+
+
+def populate_school_data(school_df: pd.DataFrame, template_path: Path, school_id, output_path=None, school_placeholder="School A") -> TemplateFiller:
+    """Function to populate and save the template with: name of school and filled tables
+
+    Args: school_filter(pd.DataFrame), Path, school_id
+
+    Returns: The template filler with populated data and appropriate school name saved in the output path"""
+
+    template_filler = populate_school_tables(school_df, template_path, school_id, output_path)

```

Comment:
```suggestion
    template_filler = populate_school_tables(school_df, template_path)

```
I think this is all we need to call `populate_school_tables`?